service/state/state.go

```diff
@@ -15,8 +15,11 @@ type Tx = coretypes.Tx
 // TxResponse is an alias to the TxResponse type from Cosmos-SDK.
 type TxResponse = sdk.TxResponse
 
-// Address is an alias to the Address type from Cosmos-SDK.
-type Address = sdk.Address
+// ValAddress is an alias to the ValAddress type from Cosmos-SDK.
+type ValAddress = sdk.ValAddress
+
+// AccAddress is an alias to the AccAddress type from Cosmos-SDK.
+type AccAddress = sdk.AccAddress
```

Comment:
What's the reason for this change?

Reply:
CC: @renaynay 

To make it explicit what account type is accepted for each request, instead of being hidden in the implementation

Reply:
Feel conflicted about this refactor slipping through. I like explicitness in the case where we work with ValAddress in delegation related methods. However, the Address abstraction should still be preserved for Transfer and Balance methods to allows operations for any kind of addresses!

Reply:
it doesn't though. transactions and balances were already only using AccAddresses before

Reply:
(they do AccAddressFromBech32 which only accepts user accounts)

Reply:
We doesn't, but it does not meant that we shouldn't or won't. Current abstraction allows this, this _uninvited_ refactor prevents this. Once there is a user requesting feature to get balance of validator, we will readd this. 

If we really want to make this PR complete all features:
Extending support for ValAddress via Address abstraction > refactoring the code to remove abstraction at all.

Reply:
Sorry but this was invited, it was part of rene's review. Let's discuss in Sync today.

Reply:
@Wondertan let me clarify: 

We have 4 things going on under the hood in state access: 
1. bank module writes (transfer, submitTx, submitPFD which is just fancy submitTx)
2. bank module reads (balance, balanceForAddress same thing)
3. staking module writes (Delegate, Redelegate, etc.)
4. staking module reads (queryDelegate, queryUnbonding, etc.)

1 and 2 access same account state tree via the bank store key (albeit in separate ways since we do proof verification for bank module reads)

3 and 4 access same staking state tree via staking store key (this happens via grpc query client)

There are different representations of addresses that are used as keys to read from the respective trees. IMO it's good to make it explicit for bank module reads vs staking module reads since the error that is returned when you try to use something other than AccAddress for bank module read from the app is quite ugly and difficult for the user to parse through (even though it's very low probability they run into it if they're using our key management utility).

Reply:
What we can and *should* do is clarify this in the PR title and description pls 🙏🏻 @distractedm1nd

Reply:
#1085